src/posix/platform/radio_url.hpp

```diff
@@ -52,7 +52,19 @@ class RadioUrl : public ot::Url::Url
      * @param[in]   aUrl    The null-terminated URL string.
      *
      */
-    RadioUrl(const char *aUrl);
+    explicit RadioUrl(const char *aUrl) { Init(aUrl); };
+
+    /**
+     * Initializes the radio URL.
+     *
+     * @param[in]   aUrl    The null-terminated URL string.
+     *
+     */
+    void Init(const char *aUrl);
+
+    RadioUrl(RadioUrl &)       = delete;
+    RadioUrl(const RadioUrl &) = delete;
+    RadioUrl(RadioUrl &&)      = delete;
```

Comment:
```suggestion
    RadioUrl(const RadioUrl &) = delete;
```

The deletion of copy constructor already prevents the implicit definition of move constructor.

Reply:
Updated